spot-client/src/spot-remote/ui/views/remote-views/waiting-for-call.js

```diff
@@ -160,10 +187,55 @@ export default class WaitingForCallView extends React.PureComponent {
     /**
      * Updates the state to display screen share.
      *
+     * @param {boolean} wireless - True for wireless or false for wired.
      * @private
      * @returns {void}
      */
-    _onSetShareContentActive() {
+    _onJoinWithScreensharing(wireless) {
+        // This prevents from triggering the action twice
+        if (this._goToMeetingPromise) {
+            return;
+        }
+
         this.setState({ activeTab: 'share' });
+
+        const goToMeetingPromise = this.props.onGoToMeeting(
```

Comment:
Todo for another day: we should research cancellable promises. I've been using some _isMounted/isUnMounted hack.